cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/IRConstruction.qll

```diff
@@ -35,29 +36,41 @@ module Raw {
   cached
   predicate functionHasIR(Function func) { exists(getTranslatedFunction(func)) }
 
+  cached
+  predicate varHasIRFunc(GlobalOrNamespaceVariable var) {
+    var.hasInitializer() and
+    (
+      not var.getType().isDeeplyConst()
```

Comment:
This, I assume, is to avoid some of the pref problems you were running into? Maybe worth a comment?

Reply:
Mostly that constant integers will end up getting inlined in IR generation, so we don't need IR for them.